pdc/apps/package/serializers.py

```diff
@@ -107,48 +108,74 @@ class Meta:
                   'composes', 'subvariant')
 
 
+def _get_fields(data, fields):
+    """
+    Check if all listed fields are present in data and return a dict containing
+    just those.
+
+    When something is missing, ValidationError is raised.
+    """
+    required_data = {}
+    errors = {}
+    for field in fields:
+        try:
+            required_data[field] = data[field]
+        except KeyError:
+            errors[field] = 'This field is required.'
+    if errors:
+        raise serializers.ValidationError(errors)
+    return required_data
+
+
+def _log_new_object(request, obj, route, topic, data):
```

Comment:
Would be better to split this into two functions, one creating changeset and other the message.